starter-code/app.js

```diff
@@ -1,5 +1,61 @@
 const express = require('express');
 const app = express();
+const expressLayouts = require('express-ejs-layouts');
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
 
+// -- setup
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+
+// -- routes
+
+app.get('/random', (req, res, next) => {
+        // Retrieve a random chuck joke
+        next("SDFAFSDFSDFSDF");
+    client.getRandomJoke()
+    .then((response) => {
+        res.render('index',response);
+    }).catch((err) => {
+        next(err);
+    });
+});
+  
+app.get('/categories', (req, res, next) => {
+//    console.log(req.query);
+    client.getJokeCategories()
+    .then((response)=>  {
+        const data = {
+            categories: response
+        };
+        res.render('categories', data);
+    })
+    .catch((err)=> {
+        next(err);
+    });
+});
+
+app.get('/categories/:cat', (req, res) => {
+    console.log(req.params.cat);
+    res.send('GET');
```

Comment:
Here you can use a method of the API for returning a random joke of that category and render it in a page (joke-by-category in the case above):
```
client.getRandomJoke(category).then(function (joke) {
  res.render('joke-by-category', {joke: joke.value, category});
});
```